<!-- To be run on the Raspberry Pi, not the master machine -->
<launch>

  <!-- Serves as camera identifier and topic/service namespace -->
  <arg name="node_name" default="picam1" />

  <!-- Output Parameters -->
  <arg name="save_local" default="true" />
  <arg name="save_dir" default="$(find ros_picam)/captures" />
  <arg name="publish" default="true" />
  <arg name="publish_topic" default="picam_output" />

  <!-- Camera Parameters -->
  <!-- Primary usage is for still image collection/evaluation, video services still need work. -->
  <arg name="enable_video" default="false" />
  <!-- Camera Object Configuration -->
  <arg name="width" default="3280" /> <!-- Max video: 1920, Max still: 3280 -->
  <arg name="height" default="2464" /> <!-- Max video: 1080, Max still: 2464 -->
  <arg name="fps" default="30" />
  <arg name="iso" default="100" /> <!-- Range: 0 to 1600 -->
  <arg name="brightness" default="50" /> <!-- Range: 0 to 100 -->
  <arg name="contrast" default="0" /> <!-- Range: -100 to 100 -->
  <arg name="rotation" default="180" /> <!-- Accepts: 0, 90, 180, 270 -->


  <!-- Run Picam Node -->
  <node name="$(arg node_name)"
    pkg="ros_picam"
    type="picam_client_v2.py"
    output="screen"
    respawn="true"
    respawn_delay="5">
    <!-- Output Parameters -->
    <param name="save_local" value="$(arg save_local)" />
    <param name="save_dir" value="$(arg save_dir)" />
    <param name="publish" value="$(arg publish)" />
    <param name="publish_topic" value="$(arg publish_topic)" />
    <!-- Camera Parameters -->
    <param name="enable_video" value="$(arg enable_video)" />
    <param name="im_width" value="$(arg width)" />
    <param name="im_height" value="$(arg height)" />
    <param name="fps" value="$(arg fps)" />
    <param name="iso" value="$(arg iso)" />
    <param name="brightness" value="$(arg brightness)" />
    <param name="contrast" value="$(arg contrast)" />
    <param name="rotation" value="$(arg rotation)" />
  </node>

</launch>
